numpy/ma/core.py

```diff
@@ -1613,7 +1613,11 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
             else:
                 result = m
         else:
-            result = np.array(m, dtype=dtype, copy=copy)
+            # existence of fields implies True
+            if m.dtype.fields and np.dtype(dtype) == np.bool_:
+                result = np.ones(m.shape, dtype=dtype)
```

Comment:
This was necessary in order to preserve the old behavior and get the MA tests to pass, at the time I wrote it. It is possible that with the recent changes to structured masks you've worked on this can be updated.

More specifically, in this PR I removed a special case in the C code [here](https://github.com/numpy/numpy/pull/6053/files#diff-d3108a3599564ada8ce08fe45f539350L2650) since I totally rewrote the casting code for structs. 

That block is a special case that converts strucs to booleans, as in:
```python
>>> a = np.zeros(4, dtype='i4,f4')
>>> a.astype(bool)
array([ True,  True,  True,  True], dtype=bool)
```
I removed it because it didn't fit in with my changes and seemed like a weird hack.

However that behavior was necessary in MA code somewhere, so I had to patch up the MA code as in the the line you commented on.